### PR TITLE
fix(Core/DB): Quest 'Inoculation'

### LINKED ISSUE #5917: Quest item for "Inoculation" completes objectives when casting is cancelled
CC original issue : https://github.com/chromiecraft/chromiecraft/issues/663

#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [x] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

During the quest Inoculation (https://wowgaming.altervista.org/aowow/?quest=9303) on Azuremyst Isle,
when casting the quest item (https://wowgaming.altervista.org/aowow/?item=22962) you can cancel the channel before it is complete, but the quest objective will be completed regardless.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Cancelling the quest item channeling before completion should not still complete the quest objective.

##### STEPS TO REPRODUCE THE PROBLEM:

1. Get the quest Inoculation in Azuremyst Isle (quest id: 9303)
2. Use the quest item Inoculating Crystal on a Nestlewood Owlkin (npc id: 16518)
3. Cancel the channeling before it is complete.
4. Wait 2-3 seconds. The npc will be inoculated as if you finished the channeling.

##### EXTRA NOTES:

This is my first bug report, let me know if anything is incorrect and I will clean it up.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/0d699222de92185681f9a78822a0e5d9898294f4

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/98542582-quest-item-for-inoculation-completes-objectives-when-casting-is-cancelled?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by hanabi5**
## Changes Proposed:
- Removed SmartAI from Nestlewood Owlkin
- Scripted spell for inoculation

## Issues Addressed:

- Closes #5917
- Closes https://github.com/chromiecraft/chromiecraft/issues/663

## SOURCE:

[youtube](https://www.youtube.com/watch?v=a_qx4WRD27U&ab_channel=Bue)

## Tests Performed:

- Tested in-game

## How to Test the Changes:

1. .go c id 16518
2. .q a 9303
3. Use quest item on Nestlewood Owlkins to complete the quest.

## Known Issues and TODO List:

None

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Nyeriah** [comment]
Thanks for the PR but I have an objection to this, no spell script should ever need to handle creature AI that way. What was the issue with the SAI script the creature had before?

**Comment by hanabi5** [comment]
There's no smart hook to handle periodic aura, only spell hit. If the creature gets hit by the spell, it triggers the smartai, but it should only trigger after aura is applied.
Should I make a new function to handle aura just for one creature?

File: data/sql/updates/pending_db_world/rev_1654292539705612400.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+SET @NESTLEWOOD_OWLKIN  = 16518;
+SET @SPELL_INOCULATE    = 29528;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1654292539705612400.sql:2]:
```suggestion
SET @NESTLEWOOD_OWLKIN  := 16518;
SET @SPELL_INOCULATE   := 29528;
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/scripts/Spells/spell_generic.cpp [hunk 2]
```diff
@@ -4514,6 +4514,39 @@ class spell_gen_remove_impairing_auras : public SpellScript
...
+                owlkin->SetUnitFlag(UnitFlags(UNIT_FLAG_IMMUNE_TO_PC));
+                owlkin->DespawnOrUnsummon(15000);
+                caster->RewardPlayerAndGroupAtEvent(INOCULATED_OWLKIN_ENTRY, caster);
+            }
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_inoculate_nestlewood_owlkin::HandleEffectPeriodic, EFFECT_0, SPELL_AURA_PERIODIC_TRIGGER_SPELL);
```
Comment (Kitzunu) [inline src/server/scripts/Spells/spell_generic.cpp:4546]: Effect 0: Id 6 (SPELL_EFFECT_APPLY_AURA) not SPELL_AURA_PERIODIC_TRIGGER_SPELL

Comment (Kitzunu) [inline src/server/scripts/Spells/spell_generic.cpp:4546]:
```suggestion
        OnEffectPeriodic += AuraEffectPeriodicFn(spell_inoculate_nestlewood_owlkin::HandleEffectPeriodic, EFFECT_0, SPELL_EFFECT_APPLY_AURA);
```

**Comment by hanabi5** [comment]
I mean, the script works it's just that it doesn't pass the DB checks

File: src/server/scripts/Spells/spell_generic.cpp [hunk 3]
```diff
@@ -4514,6 +4514,39 @@ class spell_gen_remove_impairing_auras : public SpellScript
     }
 };
 
+enum AzuremistIsles
+{
+    NESTLEWOOD_OWLKIN_ENTRY = 16518,
+    INOCULATED_OWLKIN_ENTRY = 16534,
+};
+
+class spell_inoculate_nestlewood_owlkin : public AuraScript
```
Comment (Kitzunu) [inline src/server/scripts/Spells/spell_generic.cpp:4523]:
```suggestion
class spell_inoculate_nestlewood_owlkin : public SpellScript
```

**Comment by Kitzunu** [comment]
I am trying to fix the build, after I was asked. kek

**Comment by acidmanifesto** [comment]
For those curious about this. There WAS a cmake issue on the master that was resolved yesterday dealing with sql. As for the AITarget error it was a missing include directive.

**Pull request closed without merging**